### PR TITLE
Document process for bumping dev-master branch-alias and bump to 1.10-dev

### PR DESCRIPTION
**Pull request by jmikola**
This was missed after the 1.10.0 release.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -124,9 +124,38 @@ The maintenance branch's HEAD will be the target for our release tag:
 $ git tag -a -m "Release X.Y.Z" X.Y.Z
 ```
 
-### Push tags
+### Update branch-alias for dev-master (major and minor releases only)
```
Comment (jmikola) [inline CONTRIBUTING.md:127]: @tanlisu: See [branch-alias](https://getcomposer.org/doc/articles/aliases.md#branch-alias) for additional context.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -124,9 +124,38 @@ The maintenance branch's HEAD will be the target for our release tag:
...
+
+Initial versions of new major or minor releases will likely be made on the
+master branch directly instead of a maintenance branch. In this case, the
+`branch-alias` for `dev-master` in `composer.json` should be updated. For
+example, after tagging a new 1.10.0 release on the master branch,
+`composer.json` in the master branch may still read:
```
Comment (alcaeus) [inline CONTRIBUTING.md:133]: Regardless of when releases are tagged, the `branch-alias` needs to be changed once we branch off. For example, after branching `v1.10`, the `branch-alias` for `dev-master` should be changed to `1.11.x`.

**Review by alcaeus — CHANGES_REQUESTED** [review]

File: CONTRIBUTING.md (hunk 2 above)
Comment (jmikola) [reply CONTRIBUTING.md:133]: Good point. I'll move this to a separate section that talks about branching in that case.

**Review by tanlisu — APPROVED** [review]
LGTM

**Review by alcaeus — APPROVED** [review]
Thanks!

**Pull request merged**